include/deal.II/base/template_constraints.h

```diff
@@ -27,6 +27,68 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+template <class...>
+using void_t = void;
```

Comment:
Are we going to use this in public interfaces? If not, can we put it into the internal namespace as well?

Reply:
Actually, all of the things you're adding.